akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -302,6 +302,41 @@ abstract class ConnectionPoolSpec(poolImplementation: PoolImplementation) extend
       Await.result(gateway.poolStatus(), 1500.millis.dilated).get shouldBe a[PoolInterfaceRunning]
       awaitCond({ Await.result(gateway.poolStatus(), 1500.millis.dilated).isEmpty }, 2000.millis.dilated)
     }
+    "automatically shutdown after configured timeout periods but only after streaming response is finished" in new TestSetup() {
+      val (requestIn, responseOut, responseOutSub, hcp) = cachedHostConnectionPool[Int](idleTimeout = 200.millis)
+      val gateway = hcp.gateway
+
+      val responseEntityPub = TestPublisher.probe[ByteString]()
+
+      override def testServerHandler(connNr: Int): HttpRequest => HttpResponse = {
+        case request @ HttpRequest(_, Uri.Path("/a"), _, _, _) =>
+          val entity = HttpEntity.Chunked.fromData(ContentTypes.`text/plain(UTF-8)`, Source.fromPublisher(responseEntityPub))
+          super.testServerHandler(connNr)(request) withEntity entity
+        case x => super.testServerHandler(connNr)(x)
+      }
+      requestIn.sendNext(HttpRequest(uri = "/a") -> 42)
+      responseOutSub.request(1)
+      acceptIncomingConnection()
+      val (Success(r1), 42) = responseOut.expectNext()
+      val responseEntityProbe = ByteStringSinkProbe()
+      r1.entity.dataBytes.runWith(responseEntityProbe.sink)
+      responseEntityPub.sendNext(ByteString("Hello"))
+      responseEntityProbe.expectUtf8EncodedString("Hello")
+
+      Await.result(gateway.poolStatus(), 1500.millis.dilated).get shouldBe a[PoolInterfaceRunning]
+      Thread.sleep(500)
+
+      // afterwards it should still be running because a response is still being delivered
```

Comment:
Hmm, so a slow reader would hold up closing the pool. I guess that's OK, and this is probably guarded by another timeout somewhere.

Reply:
Yes, exactly, we fix #1847 with that new functionality.